diff -urN spt3g_software_orig/calibration/python/__init__.py spt3g_software/calibration/python/__init__.py
--- spt3g_software_orig/calibration/python/__init__.py	2021-09-01 14:56:55.000000000 -0700
+++ spt3g_software/calibration/python/__init__.py	2021-11-12 10:03:24.340992684 -0800
@@ -1,5 +1,4 @@
-from spt3g.core.load_pybindings import load_pybindings
-load_pybindings(__name__, __path__)
+from .libcalibration import *
 
 from . import build_cal_frames
 
diff -urN spt3g_software_orig/core/python/__init__.py spt3g_software/core/python/__init__.py
--- spt3g_software_orig/core/python/__init__.py	2021-09-01 14:56:55.000000000 -0700
+++ spt3g_software/core/python/__init__.py	2021-11-12 10:03:54.748855094 -0800
@@ -1,5 +1,4 @@
-from spt3g.core.load_pybindings import load_pybindings
-load_pybindings(__name__, __path__)
+from .libcore import *
 
 from spt3g.core.g3logging import log_trace, log_debug, log_info, log_notice, log_warn, log_error, log_fatal, set_log_level
 
diff -urN spt3g_software_orig/dfmux/python/__init__.py spt3g_software/dfmux/python/__init__.py
--- spt3g_software_orig/dfmux/python/__init__.py	2021-09-01 14:56:55.000000000 -0700
+++ spt3g_software/dfmux/python/__init__.py	2021-11-12 10:04:28.188703931 -0800
@@ -1,5 +1,4 @@
-from spt3g.core.load_pybindings import load_pybindings
-load_pybindings(__name__, __path__)
+from .libdfmux import *
 
 from .HardwareMapTools import (
     GenerateFakeHardwareMap,
diff -urN spt3g_software_orig/gcp/python/__init__.py spt3g_software/gcp/python/__init__.py
--- spt3g_software_orig/gcp/python/__init__.py	2021-09-01 14:56:55.000000000 -0700
+++ spt3g_software/gcp/python/__init__.py	2021-11-12 10:05:03.240545634 -0800
@@ -1,5 +1,4 @@
-from spt3g.core.load_pybindings import load_pybindings
-load_pybindings(__name__, __path__)
+from .libgcp import *
 
 from .ARCExtractor import UnpackACUData, UnpackTrackerData, DecryptFeatureBit, ARCExtract, ARCExtractMinimal
 from .ARCHKExtractor import UnpackSPTpolHKData
diff -urN spt3g_software_orig/maps/python/__init__.py spt3g_software/maps/python/__init__.py
--- spt3g_software_orig/maps/python/__init__.py	2021-09-01 14:56:55.000000000 -0700
+++ spt3g_software/maps/python/__init__.py	2021-11-12 10:05:35.480400166 -0800
@@ -1,5 +1,4 @@
-from spt3g.core.load_pybindings import load_pybindings
-load_pybindings(__name__, __path__)
+from .libmaps import *
 
 # Just run this, no symbols we need though
 from .skymapaddons import *
