diff -urN spt3g_software_orig/cmake/Spt3gBoostPython.cmake spt3g_software/cmake/Spt3gBoostPython.cmake
--- spt3g_software_orig/cmake/Spt3gBoostPython.cmake	2022-11-09 12:55:44.950643068 -0800
+++ spt3g_software/cmake/Spt3gBoostPython.cmake	2022-11-09 12:59:23.391502813 -0800
@@ -1,6 +1,6 @@
 # Locate Python
 if(${CMAKE_VERSION} VERSION_GREATER_EQUAL 3.12)
-	find_package(Python COMPONENTS Interpreter Development)
+	find_package(Python COMPONENTS Interpreter)
 else()
 	find_package(PythonInterp)
 	find_package(PythonLibs ${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR})
