diff -urN spt3g_software_orig/cmake/Spt3gBoostPython.cmake spt3g_software/cmake/Spt3gBoostPython.cmake
--- spt3g_software_orig/cmake/Spt3gBoostPython.cmake	2024-08-22 10:25:14.077183587 -0700
+++ spt3g_software/cmake/Spt3gBoostPython.cmake	2024-08-22 10:27:10.956608507 -0700
@@ -1,7 +1,7 @@
 # Locate Python
 
 if(${CMAKE_VERSION} VERSION_GREATER_EQUAL 3.12)
-	find_package(Python COMPONENTS Interpreter Development)
+	find_package(Python COMPONENTS Interpreter)
 else()
 	find_package(PythonInterp)
 	find_package(PythonLibs ${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR})
diff -urN spt3g_software_orig/CMakeLists.txt spt3g_software/CMakeLists.txt
--- spt3g_software_orig/CMakeLists.txt	2024-08-22 10:24:59.301256298 -0700
+++ spt3g_software/CMakeLists.txt	2024-08-23 12:33:59.073140890 -0700
@@ -42,7 +42,7 @@
 
 # Raise errors on every warning by default
 # (use target-specific options to disable particular warnings)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Werror")
+#set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Werror")
 
 # Interface library for flags and library dependencies
 add_library(spt3g INTERFACE)
diff -urN spt3g_software_orig/core/CMakeLists.txt spt3g_software/core/CMakeLists.txt
--- spt3g_software_orig/core/CMakeLists.txt	2024-08-06 11:34:45.598647939 -0700
+++ spt3g_software/core/CMakeLists.txt	2024-08-22 10:29:17.655985088 -0700
@@ -105,8 +105,8 @@
 add_spt3g_test(quaternions)
 add_spt3g_test(timesample)
 
-add_spt3g_test_program(test
-                       SOURCE_FILES
-                         ${CMAKE_CURRENT_SOURCE_DIR}/tests/G3TimestreamTest.cxx
-                         ${CMAKE_CURRENT_SOURCE_DIR}/tests/G3TimestreamMapTest.cxx
-                       USE_PROJECTS core)
+#add_spt3g_test_program(test
+#                       SOURCE_FILES
+#                         ${CMAKE_CURRENT_SOURCE_DIR}/tests/G3TimestreamTest.cxx
+#                         ${CMAKE_CURRENT_SOURCE_DIR}/tests/G3TimestreamMapTest.cxx
+#                       USE_PROJECTS core)
diff -urN spt3g_software_orig/examples/CMakeLists.txt spt3g_software/examples/CMakeLists.txt
--- spt3g_software_orig/examples/CMakeLists.txt	2024-08-06 11:34:45.610647890 -0700
+++ spt3g_software/examples/CMakeLists.txt	2024-08-22 10:56:07.300059646 -0700
@@ -1,2 +1,2 @@
-add_executable(cppexample cppexample.cxx) 
-target_link_libraries(cppexample core)
+#add_executable(cppexample cppexample.cxx) 
+#target_link_libraries(cppexample core)
Binary files spt3g_software_orig/.git/index and spt3g_software/.git/index differ
