diff -urN spt3g_software_orig/core/src/G3TimeStamp.cxx spt3g_software/core/src/G3TimeStamp.cxx
--- spt3g_software_orig/core/src/G3TimeStamp.cxx	2025-01-28 20:47:41.412059665 -0800
+++ spt3g_software/core/src/G3TimeStamp.cxx	2025-01-28 21:09:58.481129822 -0800
@@ -2,7 +2,7 @@
 #include <serialization.h>
 
 #include <stdint.h>
-#include <time.h>
+#include <sys/time.h>
 #include <sstream>
 #include <iomanip>
 
